//
// {C} Copyright 2020 Pensando Systems Inc. All rights reserved
//
//------------------------------------------------------------------------------
///
/// \file
/// protobuf specification for alerts
///
//------------------------------------------------------------------------------

syntax = "proto3";
package operd;

import "gogo.proto";
import "types.proto";

enum AlertSeverity {
  DEBUG    = 0;
  INFO     = 1;
  WARN     = 2;
  CRITICAL = 3;
}

message Alert {
  uint64        Timestamp   = 1;
  string        Name        = 2;
  string        Category    = 3;
  AlertSeverity Severity    = 4;
  string        Description = 5;
  string        Message     = 6;
}

// Alerts get response
message AlertsGetResponse {
  // success or failure status
  types.ApiStatus   ApiStatus = 1;
  Alert             Response  = 2;
}
